freud/box.pyx

```diff
@@ -462,6 +465,28 @@ cdef class Box:
         else:
             raise ValueError("Box can only be multiplied by positive values.")
 
+    def plot(self, title=None, ax=None, image=[0, 0, 0], *args, **kwargs):
+        """Plot a :class:`~.box.Box` object.
+
+        Args:
+            title (str):
+                Title of the graph. (Default value = :code:`None`).
+            ax (:class:`matplotlib.axes.Axes`):
+                Axes object to plot. If :code:`None`, make a new axes and
+                figure object. If plotting a 3D box, the axes must be 3D.
+                (Default value = :code:`None`).
+            image (list):
+                The periodic image location at which to draw the box (Default
+                value = :code:`[0, 0, 0]`).
+            ``*args``, ``**kwargs``:
+                All other arguments are passed on to
+                :meth:`mpl_toolkits.mplot3d.Axes3D.plot` or
+                :meth:`matplotlib.axes.Axes.plot`.
+        """
+        import freud.plot
```

Comment:
As discussed offline with @bdice, this is an API choice to have the `obj.plot()` methods fail if users request them and `matplotlib` is not available, but the `_repr_png` methods should not fail since a user is not necessarily actively requesting a plot, so the fallback to `__repr__` is reasonable.